### ISSUE TITLE
Else clause should be ignored on sql query builders

### ISSUE THREAD

**Issue by talhaamir4130** [phpmd/phpmd#979]
## Description
Avoiding else clauses are good but sometimes they are necessary.

For example I have a query:
```
$query = 'SELECT * FROM EMPLOYEE';
.
.
.
if ($variable) {
       $query->andWhere('e.id > 50');
} else {
       $query->andWhere('e.id < 50');
}
```
Here is not possible to add `Where` clause in query before if then check for if then add `Where` once again.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] Tell if you have the option to provide the code for this proposal.
* [x] This issue is about 1 feature proposal and nothing more.
* [x] The issue has a descriptive title. For example: "Add JSON render option".

**Comment by tvbeek** [comment]
Hello @talhaamir4130 

We (currently) don't have a way to handle exceptions while finding out that we are in a sql query builder. That should also be a very complex part. Because what are the builders we need to handle. And we need to see that the else defines a query for that object.

Personal I should extract the where statement from the call to andWhere, example:
```php
$myWhereStatement = 'e.id < 50';
if ($variable) {
       $myWhereStatement = 'e.id > 50';
}
$query->andWhere($myWhereStatement);
```

But you can also suppress the warnings, see: https://phpmd.org/documentation/suppress-warnings.html

Is this clear for you? (Then we can close this issue) or do you mean something else?

**Comment by AJenbo** [comment]
It's a bit hard to make suggestion without knowing the specific query builder.

Some generic alternatives:
```php
$query->andWhere($variable ? 'e.id > 50' : 'e.id < 50');
```

```php
$operant = $variable ? '>' : '<';
$query->andWhere("e.id $operant 50");
```

```php
private function whereRange(bool $variable): void
        if ($variable) {
            $query->andWhere('e.id > 50');
            return;
        }

    $query->andWhere('e.id < 50');
}
```

**Comment by talhaamir4130** [comment]
@tvbeek @AJenbo 
These solutions are good and I already do these, but I was just wondering if it could be implemented.

It seems it's complicated to implement so I am closing this issue. Thanks
